include/dev/HallSensor.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef HALLSENSOR_HPP
+#define HALLSENSOR_HPP
+
+#include <EVT/io/GPIO.hpp>
+
+namespace IO = EVT::core::IO;
+
+namespace WSS::DEV {
+
+/**
+ * This is the class for each individual front and back HallSensor
+ */
+class HallSensor {
+public:
+    /** The state of the wheel whether it is stopped, starting to spin, or spinning */
+    enum class WheelSpeedState {
+        STOP,         /** Wheel is not moving */
+        INITIALIZING, /** Setting speed based on first reading */
+        MAINTAIN,     /** Wheel is spinning at a constant speed or speeding up */
```

Comment:
Could be constant, speeding up, or slowing down
```suggestion
        MAINTAIN,     /** Wheel is spinning */
```